File: 006Best_sightseeing_combination/006Best_sightseeing_combination.cpp
#include<vector>
#include<iostream>
#include<algorithm>

using namespace std;

int maxScoreSightseeingPair(vector<int>& A);

int main() {
	vector<int> list{ 8,1,5,2,6 };
	cout << "result:" << maxScoreSightseeingPair(list) << endl;
	return 0;
}

int maxScoreSightseeingPair(vector<int>& A) {
	int max_i = 0;
	int max_iaddj = 0;
	if (A.size() != 0)
		max_i = A[0];
	else return 0;
	for (int j = 1;j < A.size();j++)
	{
		max_i = max(max_i, A[j]);
		max_iaddj = max(max_iaddj, max_i + A[j] - j);
	}
	return max_iaddj;
}

File: 006Best_sightseeing_combination/006Best_sightseeing_combination.py
def maxScoreSightseeingPair(A: [int]) -> int:
    max_i = 0
    max_iaddj = 0
    if len(A) == 0:
        return 0
    else:
        max_i = A[0]
    for i in range(1,len(A)):
        max_i = max(max_i, A[i])
        max_iaddj = max(max_iaddj, max_i + A[i] - i)
    return max_iaddj

my_list = [8, 1, 5, 2, 6]
print(maxScoreSightseeingPair(my_list))